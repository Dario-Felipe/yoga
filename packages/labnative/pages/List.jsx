import React from 'react';
import { Alert } from 'react-native';
import { Text, List } from '@gympass/yoga';

const listItemData = [
  { key: 'Devin' },
  { key: 'Dan' },
  { key: 'Dominic' },
  { key: 'Jackson' },
  { key: 'James' },
];

const listTouchableItemData = [
  { key: 'Joel' },
  { key: 'John' },
  { key: 'Jillian' },
  { key: 'Jimmy' },
  { key: 'Julie' },
];

const ListPage = () => {
  return (
    <>
      <Text.SectionTitle>List Item</Text.SectionTitle>
      <List
        data={listItemData}
        renderItem={({ item }) => (
          <List.Item>
            <Text>{item.key}</Text>
          </List.Item>
        )}
      />

      <Text.SectionTitle>List Touchable Item</Text.SectionTitle>
      <List
        scrollEnabled={false}
        data={listTouchableItemData}
        renderItem={({ item }) => (
          <List.TouchableItem
            small
            onPress={() => Alert.alert('You pressed', item.key)}
          >
            <Text>{item.key}</Text>
          </List.TouchableItem>
        )}
      />
    </>
  );
};

export default ListPage;
