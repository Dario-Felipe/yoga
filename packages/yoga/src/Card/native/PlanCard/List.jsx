import React from 'react';
import styled, { css, withTheme } from 'styled-components';
import { View, TouchableWithoutFeedback } from 'react-native';
import { string, node, shape, oneOfType, func } from 'prop-types';

import Text from '../../../Text';
import theme from '../../../Theme/helpers/themeReader';

const List = styled.View`
  margin-top: ${theme.components.card.plan.list.margin.top}px;
`;

const Item = styled.View`
  margin-bottom: ${theme.components.card.plan.list.item.margin.bottom}px;
`;

const ItemText = styled(Text.Small)`
  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      font-size: ${plan.list.item.font.size}px;
      color: ${plan.list.item.font.color};
    `;
  }}
`;

const Button = styled.View`
  margin-top: ${theme.components.card.plan.list.button.margin.top}px;
`;

const ButtonText = styled(Text.Medium)`
  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      font-size: ${plan.list.button.font.size}px;
      color: ${plan.list.button.font.color};
    `;
  }}
`;

const ListItem = withTheme(
  ({
    text,
    icon: Icon,
    buttonProps: { children, ...buttonProps },
    theme: yogaTheme,
  }) => (
    <Item>
      <View style={{ flexDirection: 'row' }}>
        {Icon && (
          <Icon
            width={16}
            height={16}
            style={{
              marginRight:
                yogaTheme.yoga.components.card.plan.list.item.icon.margin.right,
            }}
          />
        )}
        <ItemText>{text}</ItemText>
      </View>
      {Boolean(Object.keys(buttonProps).length) && (
        <TouchableWithoutFeedback {...buttonProps}>
          <Button>
            <ButtonText>{children}</ButtonText>
          </Button>
        </TouchableWithoutFeedback>
      )}
    </Item>
  ),
);

List.displayName = 'PlanCard.List';
ListItem.displayName = 'PlanCard.ListItem';
Button.displayName = 'PlanCard.ListButton';

ListItem.propTypes = {
  text: string.isRequired,
  /** an icon to be displayed on the begin of the item */
  icon: oneOfType([node, func]),
  /** if provided displays a button below the item text. It accepts all button
   * element props */
  buttonProps: shape({}),
};

ListItem.defaultProps = {
  icon: undefined,
  buttonProps: {},
};

export { List, ListItem };
