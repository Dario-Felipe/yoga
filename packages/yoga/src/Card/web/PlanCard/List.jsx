import React from 'react';
import styled, { css } from 'styled-components';
import { string, node, shape, oneOfType, func } from 'prop-types';

import theme from '../../../Theme/helpers/themeReader';

const truncateStyle = css`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const List = styled.ul`
  margin: ${theme.components.card.plan.list.margin.top}px 0 0;
  padding: 0;

  list-style: none;
`;

const Item = styled.li`
  ${truncateStyle}

  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      margin-bottom: ${plan.list.item.margin.bottom}px;

      font-size: ${plan.list.item.font.size}px;
      color: ${plan.list.item.font.color};

      svg {
        margin-right: ${plan.list.item.icon.margin.right}px;
      }
    `;
  }}
`;

const Button = styled.button`
  display: block;
  padding: 0;

  letter-spacing: normal;

  background-color: transparent;
  border: none;

  cursor: pointer;
  text-decoration: none;

  ${truncateStyle}

  ${props => {
    const {
      components: {
        card: { plan },
      },
    } = theme(props);

    return css`
      margin-top: ${plan.list.button.margin.top}px;

      font-family: ${theme.baseFont.family};
      font-size: ${plan.list.button.font.size}px;
      font-weight: ${plan.list.button.font.weight};
      color: ${plan.list.button.font.color};
    `;
  }}
`;

const ListItem = ({ text, icon: Icon, buttonProps }) => (
  <Item>
    {Icon && <Icon width={16} height={16} />}
    {text}
    {Object.keys(buttonProps).length && <Button {...buttonProps} />}
  </Item>
);

List.displayName = 'PlanCard.List';
ListItem.displayName = 'PlanCard.ListItem';
Button.displayName = 'PlanCard.ListButton';

ListItem.propTypes = {
  text: string.isRequired,
  /** an icon to be displayed on the begin of the item */
  icon: oneOfType([node, func]),
  /** if provided displays a button below the item text. It accepts all button
   * element props */
  buttonProps: shape({}),
};

ListItem.defaultProps = {
  icon: undefined,
  buttonProps: {},
};

export { List, ListItem };
