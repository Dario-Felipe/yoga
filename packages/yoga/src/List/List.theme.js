const List = ({ spacing, borders, colors }) => ({
  listItem: {
    padding: {
      top: spacing.medium,
      right: spacing.large,
      bottom: spacing.medium,
      left: spacing.large,
    },
  },
  border: {
    width: borders.small,
    color: colors.gray[3],
  },
});

export default List;
