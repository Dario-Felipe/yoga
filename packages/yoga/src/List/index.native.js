import { List, ListItem, ListTouchableItem } from './native';

List.Item = ListItem;
List.TouchableItem = ListTouchableItem;

export default List;
