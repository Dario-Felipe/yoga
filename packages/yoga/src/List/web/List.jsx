import React from 'react';
import PropTypes, { string } from 'prop-types';
import styled from 'styled-components';

const StyledList = styled.ul`
  display: flex;

  width: 100%;
  padding: 0;
  margin: 0;

  list-style: none;

  ${({
    divided,
    horizontal,
    theme: {
      yoga: {
        components: { list },
      },
    },
  }) => `
    flex-direction: ${horizontal ? 'row' : 'column'};

    > * {
        ${
          divided
            ? `border-${horizontal ? 'right' : 'bottom'}: ${
                list.border.width
              }px solid ${list.border.color}`
            : ''
        };

        &:last-child {
          border: none;
        }
      }
    }
  `}
`;

/** Lists are a continuous group of text or images. They are composed of items
containing primary and supplemental actions, which are represented by icons and
text. */
const List = React.forwardRef(
  ({ horizontal, divided, testId, ...rest }, ref) => (
    <StyledList
      horizontal={horizontal}
      divided={divided}
      ref={ref}
      data-testid={testId}
      {...rest}
    />
  ),
);

List.propTypes = {
  horizontal: PropTypes.bool,
  divided: PropTypes.bool,
  testId: string,
};

List.defaultProps = {
  horizontal: false,
  divided: true,
  testId: undefined,
};

List.displayName = 'List';

export default List;
