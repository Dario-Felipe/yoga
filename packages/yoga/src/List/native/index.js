import ListItem from './ListItem';
import ListTouchableItem from './ListTouchableItem';
import List from './List';

export { List, ListItem, ListTouchableItem };
