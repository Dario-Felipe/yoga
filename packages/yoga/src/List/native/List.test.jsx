import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Text } from 'react-native';
import ThemeProvider from '../../Theme';
import List from '..';

const data = [
  { key: 'Devin' },
  { key: 'Dan' },
  { key: 'Dominic' },
  { key: 'Jackson' },
  { key: 'James' },
  { key: 'Joel' },
  { key: 'John' },
  { key: 'Jillian' },
  { key: 'Jimmy' },
  { key: 'Julie' },
];

describe('<List />', () => {
  describe('Snapshots', () => {
    describe('Without props', () => {
      it('should match snapshot with a default list', () => {
        const { container } = render(
          <ThemeProvider>
            <List
              data={data}
              renderItem={({ item }) => (
                <List.Item>
                  <Text>{item.key}</Text>
                </List.Item>
              )}
            />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with a small list', () => {
        const { container } = render(
          <ThemeProvider>
            <List
              data={data}
              renderItem={({ item }) => (
                <List.Item small>
                  <Text>{item.key}</Text>
                </List.Item>
              )}
            />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with a list without divisors', () => {
        const { container } = render(
          <ThemeProvider>
            <List
              data={data}
              renderItem={({ item }) => (
                <List.Item divided={false}>
                  <Text>{item.key}</Text>
                </List.Item>
              )}
            />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with a horizontal list', () => {
        const { container } = render(
          <ThemeProvider>
            <List
              horizontal
              data={data}
              renderItem={({ item }) => (
                <List.Item>
                  <Text>{item.key}</Text>
                </List.Item>
              )}
            />
          </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should call onPress when touchable item is pressed', () => {
        const a = { key: 'Devin', onPress: jest.fn() };
        const { getByText } = render(
          <ThemeProvider>
            <List
              data={[a]}
              renderItem={({ item }) => (
                <List.TouchableItem onPress={item.onPress}>
                  <Text>{item.key}</Text>
                </List.TouchableItem>
              )}
            />
          </ThemeProvider>,
        );

        const listTouchableItem = getByText(a.key);
        fireEvent.press(listTouchableItem);
        expect(a.onPress).toHaveBeenCalled();
      });
    });
  });
});
