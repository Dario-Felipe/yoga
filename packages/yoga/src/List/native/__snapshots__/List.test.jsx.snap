// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Snapshots Without props should match snapshot with a default list 1`] = `
<RCTScrollView
  data={
    [
      {
        "key": "Devin",
      },
      {
        "key": "Dan",
      },
      {
        "key": "Dominic",
      },
      {
        "key": "Jackson",
      },
      {
        "key": "James",
      },
      {
        "key": "Joel",
      },
      {
        "key": "John",
      },
      {
        "key": "Jillian",
      },
      {
        "key": "Jimmy",
      },
      {
        "key": "Julie",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  style={
    [
      {
        "width": "100%",
      },
    ]
  }
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Devin
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dan
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dominic
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jackson
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          James
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Joel
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          John
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jillian
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jimmy
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Julie
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<List /> Snapshots Without props should match snapshot with a horizontal list 1`] = `
<RCTScrollView
  data={
    [
      {
        "key": "Devin",
      },
      {
        "key": "Dan",
      },
      {
        "key": "Dominic",
      },
      {
        "key": "Jackson",
      },
      {
        "key": "James",
      },
      {
        "key": "Joel",
      },
      {
        "key": "John",
      },
      {
        "key": "Jillian",
      },
      {
        "key": "Jimmy",
      },
      {
        "key": "Julie",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  style={
    [
      {
        "width": "100%",
      },
    ]
  }
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Devin
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dan
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dominic
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jackson
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          James
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Joel
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          John
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jillian
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jimmy
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Julie
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<List /> Snapshots Without props should match snapshot with a list without divisors 1`] = `
<RCTScrollView
  data={
    [
      {
        "key": "Devin",
      },
      {
        "key": "Dan",
      },
      {
        "key": "Dominic",
      },
      {
        "key": "Jackson",
      },
      {
        "key": "James",
      },
      {
        "key": "Joel",
      },
      {
        "key": "John",
      },
      {
        "key": "Jillian",
      },
      {
        "key": "Jimmy",
      },
      {
        "key": "Julie",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  style={
    [
      {
        "width": "100%",
      },
    ]
  }
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          [
            {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Devin
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          [
            {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dan
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          [
            {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dominic
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          [
            {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jackson
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          [
            {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          James
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          [
            {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Joel
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          [
            {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          John
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          [
            {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jillian
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          [
            {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jimmy
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          [
            {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Julie
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<List /> Snapshots Without props should match snapshot with a small list 1`] = `
<RCTScrollView
  data={
    [
      {
        "key": "Devin",
      },
      {
        "key": "Dan",
      },
      {
        "key": "Dominic",
      },
      {
        "key": "Jackson",
      },
      {
        "key": "James",
      },
      {
        "key": "Joel",
      },
      {
        "key": "John",
      },
      {
        "key": "Jillian",
      },
      {
        "key": "Jimmy",
      },
      {
        "key": "Julie",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  style={
    [
      {
        "width": "100%",
      },
    ]
  }
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Devin
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Dan
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Dominic
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Jackson
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          James
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Joel
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          John
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Jillian
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Jimmy
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          [
            {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Julie
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
